### PR TITLE
Add the remaining algorithms in the GDAL Vector geoprocessing group

### PR DESCRIPTION
**Pull request by havatv**
Goal: Add documentation for the remaining algorithms in the GDAL Vector geoprocessing group:
"Buffer vectors", "Offset curve", "One side buffer" and "Points along lines".

Ticket(s): #

- [x] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst [hunk 1]
```diff
@@ -239,4 +353,252 @@ Outputs
...
+      - ``INPUT``
+      - [vector: line]
+      - The input line layer
+   *  - **Geometry column name**
+      - ``GEOMETRY``
+      - [string]
+        
+        Default: 'geometry'
+      - The name of the input layer geometry column to use
+   *  - **Offset distance (left-sided: positive, right-sided: negative**
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]:
```suggestion
   *  - **Offset distance (left-sided: positive, right-sided: negative)**
```

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst [hunk 2]
```diff
@@ -239,4 +353,252 @@ Outputs
...
+   :stub-columns: 0
+
+   *  - Label
+      - Name
+      - Type
+      - Description
+   *  - **Offset curve**
+      - ``OUTPUT``
+      - [vector: line]
+      - The output offset curve
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]:
```suggestion
      - The output offset curve layer
```

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst [hunk 3]
```diff
@@ -239,4 +353,252 @@ Outputs
...
+      - [tablefield: any]
+
+        Default: None
+      - Field to use for dissolving
+   *  - **Dissolve results**
+      - ``DISSOLVE``
+      - [boolean]
+
+        Default: False
+      -
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]: same as the other dissolve option?

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst [hunk 4]
```diff
@@ -239,4 +353,252 @@ Outputs
...
+
+   *  - **Dissolve by attribute**
+
+        Optional
+      - ``FIELD``
+      - [tablefield: any]
+
+        Default: None
+      - Field to use for dissolving
+   *  - **Dissolve results**
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]:
```suggestion
   *  - **Dissolve all results**
```

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst [hunk 5]
```diff
@@ -8,6 +8,99 @@ Vector geoprocessing
...
+      - ``DISSOLVE``
+      - [boolean]
+
+        Default: False
+      -
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]:
```suggestion
      - If set, the result is dissolved. If no field is set for dissolving,
        all the buffers are dissolved into one feature.
```

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst [hunk 6]
```diff
@@ -239,4 +353,252 @@ Outputs
       - ``OUTPUT``
       - [same as input]
       - The output multipart geometry layer (with dissolved geometries)
+      
+      
+.. _gdaloffsetcurve:
+
+Offset curve
+------------
+
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]:
```suggestion
Offsets lines by a specified distance.
Positive distances will offset lines to the left, and negative distances will
offset them to the right.

```

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst (hunk 3 above)
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]:
```suggestion
      - If set, the result is dissolved. If no field is set for dissolving,
        all the buffers are dissolved into one feature.
--



```

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst [hunk 7]
```diff
@@ -8,6 +8,99 @@ Vector geoprocessing
...
+        Default: 10.0
+      - Minimum: 0.0
+   *  - **Dissolve by attribute**
+
+        Optional
+      - ``FIELD``
+      - [tablefield: any]
+
+        Default: None
+      - Field to use for dissolving
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]: Done

File: source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/vectorgeoprocessing.rst]: Done

**Comment by havatv** [comment]
Thanks a lot for helping, @DelazJ!

**Pull request merged**